### ISSUE TITLE
Server Validation Errors not displaying in Templates

### ISSUE THREAD

**Issue by HashBug** [ember-cli/ember-cli#6366]
When we try to save record and API responds with a validation error and status code 422 , the [docs](http://emberjs.com/api/data/classes/DS.Errors.html) say that the errors are populated in model and can be used to display validation error messages returned from the server when a record.save() rejects. **But either the model doesn't populate automatically in my case or maybe the model content doesn't get refreshed**.
This is the content of routes file with register action that saves user to the API
```
model(){
},
actions:{
register: function(user){  
    var user = this.store.createRecord('user',user);  
    user.save().then(function(){  
        //handle success  
     },function(error){  
       //handle error  
     });  
 }); 
}
```

The error in response looks like:
```
{    
   "errors":[  
      {  
         "detail":"can't be blank",
         "source":{  
            "pointer": "/data/attributes/username"
         }
      }
   ]
} 
```
When I am accessing the errors in template as shown, nothing is displayed. I tried logging the error messages and got nothing in console.
```
<label>Username: {{input value=username}} </label>
{{#each model.errors.username as |error|}}
  <div class="error">
    {{error.message}}
    {{log error}}
  </div>
{{/each}}

{{#each model.errors.messages as |message|}}
  <div class="error">
    {{message}}
  </div>
{{/each}}
```
However when populated errors manually into the model as shown below, I got the desired result and errors successfully shown in the template.
```
_model:{}
model(){
    return this.get("_model");
},
actions:{
register: function(user){  
    var route = this;
    var user = this.store.createRecord('user',user);  
    user.save().then(function(){  
        //handle success  
     },function(){  
       //handle error  
       var errors = user.get('errors');
       route.set("_model.errors",errors);
     });  
 }); 
}
```
Is it some issue that the errors doesn't get populated in the model automatically? I tried it in Ember-cli 2.4.3, ember-data 2.4.2 and Ember-cli 2.8.0 with both RESAdapter and JSONAPIAdapter
 ---
Output from ember version --verbose && npm --version:
```
ember-cli: 2.4.3
http_parser: 2.7.0
node: 4.6.1
v8: 4.5.103.37
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 56.1
modules: 46
openssl: 1.0.2j
os: linux x64
2.15.9
```

**Comment by twokul** [comment]
I doubt it is an issue with `ember-cli`. StackOverflow might be more appropriate venue to ask that sort of questions (and you will probably receive the answer faster).

**Comment by HashBug** [comment]
Yeah I did the same 4 days ago, I posted my question on Stackoverflow as well, but It is not answered yet, that's why I considered taking this issue to the github.
[Stackoverlow Link](http://stackoverflow.com/questions/40081844/server-side-errors-not-populated-in-ember-model-with-restadapter)
However I have already solved the problem by setting the model in the action.
The [docs](http://emberjs.com/api/data/classes/DS.Errors.html) say that Ember.model has an errors property. 
When the API responds with an error, the errors property is populated but is not available in the templates. To get errors in templates, we have to set the model in the route/action. 
If we have to set the model, why don't the docs mention anything about it?

Or am I doing it in wrong way??
Shouldn't the model update/ reset itself when it gets error in response from the API as it happens in case of successful creation/updation of records?
